packages/react-instantsearch-hooks/scripts/transforms/ris-v6-to-v7.js

```diff
@@ -0,0 +1,273 @@
+    ariaPage: 'pageItemAriaLabel',
+  },
+  RangeInput: {
+    submit: 'submitButtonText',
+    separator: 'separatorElementText',
+  },
+  RefinementList: {
+    noResults: 'noResultsText',
+    submitTitle: 'submitButtonTitle',
+    resetTitle: 'resetButtonTitle',
+    showMore: 'showMoreButtonText',
+  },
+  SearchBox: {
+    submitTitle: 'submitButtonTitle',
+    resetTitle: 'resetButtonTitle',
+  },
+};
+
+const componentsWithDefaultRefinement = [
+  'HierarchicalMenu',
+  'HitsPerPage',
+  'Menu',
+  'Pagination',
+  'RangeInput',
+  'RefinementList',
+  'SearchBox',
+  'SortBy',
+  'ToggleRefinement',
+];
+
+export default function transform(file, api, options) {
+  const j = api.jscodeshift;
+  const printOptions = options.printOptions || {
+    quote: 'single',
+    shouldAddParens: false,
+  };
+  const root = j(file.source);
+  const jsxElements = root.findJSXElements();
+
+  const replaceImports = (from, to) =>
+    root
+      .find(j.ImportDeclaration)
+      .filter(
+        (path) =>
+          path.node.source.value === from ||
+          path.node.source.value.startsWith(`${from}/`)
+      )
+      .forEach((sourceImport) => {
+        j(sourceImport).replaceWith(
+          j.importDeclaration(
+            sourceImport.node.specifiers,
+            j.stringLiteral(sourceImport.value.source.value.replace(from, to))
+          )
+        );
+      });
+
+  const replacePropName = ({ element, from, to }) =>
+    jsxElements
+      .filter((p) => elementName(p) === element)
+      .find(j.JSXAttribute)
+      .filter((attribute) => propName(attribute) === from)
+      .forEach((attribute) =>
+        j(attribute).replaceWith(
+          j.jsxAttribute(j.jsxIdentifier(to), attribute.node.value)
+        )
+      );
+
+  const replaceClearsQuery = () =>
+    jsxElements
+      .filter(
+        (path) =>
+          elementName(path) === 'ClearRefinements' ||
+          elementName(path) === 'CurrentRefinements'
+      )
+      .find(j.JSXAttribute)
+      .filter((attribute) => propName(attribute) === 'clearsQuery')
+      .forEach((attribute) =>
+        j(attribute).replaceWith(
+          j.jsxAttribute(
+            j.jsxIdentifier('excludedAttributes'),
+            j.jsxExpressionContainer(j.template.expression`[]`)
+          )
+        )
+      );
+
+  const replaceTranslations = () =>
+    jsxElements
+      .filter((path) => Object.keys(translations).includes(elementName(path)))
+      .find(j.JSXAttribute)
+      .filter((attribute) => propName(attribute) === 'translations')
+      .forEach((attribute) => {
+        const translationsDict = translations[parentElementName(attribute)];
+
+        if (j(attribute).find(j.ObjectExpression).size() === 0) {
+          attribute.node.comments = [
+            j.commentBlock(
+              ` TODO: Follow the upgrade guide to rename the keys of the \`translations\` prop
+See https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/ `
```

Comment:
We could add an identifiable keyword to make clearer that this comment comes from our Codemods

```suggestion
              ` TODO (Codemods): Follow the upgrade guide to rename the keys of the \`translations\` prop
See https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/ `
```

or something similar.